spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,579 @@
+
+        File readOnlyManifest = createExport(client, readOnlyConsumer.getUuid(), null);
+        ZipFile readOnlyExport = ExportUtil.getExportArchive(readOnlyManifest);
+        List<ZipEntry> exportDir = ExportUtil.getFilesInDirectory(EXPORT_PATH, readOnlyExport);
+        List<ZipEntry> consumerFiles = exportDir.stream()
+            .filter(e -> e.getName().contains("consumer.json"))
+            .collect(Collectors.toList());
+        assertEquals(1, consumerFiles.size());
+
+        Map<String, File> actual = ExportUtil.copyToTempFiles(readOnlyExport, consumerFiles);
+        JsonNode root = new ObjectMapper()
+            .readTree(Files.readString(actual.get(consumerFiles.get(0).getName()).toPath()));
+        assertEquals(readOnlyConsumer.getUuid(), root.get("uuid").asText());
+        assertEquals(readOnlyConsumer.getName(), root.get("name").asText());
+
+        JsonNode ownerRoot = root.get("owner");
+        assertEquals(owner.getId(), ownerRoot.get("id").asText());
+        assertEquals(owner.getKey(), ownerRoot.get("key").asText());
+        assertEquals(owner.getDisplayName(), ownerRoot.get("displayName").asText());
+    }
+
+    @Nested
+    @DisplayName("Standard Export Tests")
+    @TestInstance(Lifecycle.PER_CLASS)
+    class StandardExporter {
+        private ApiClient client;
+        private OwnerClient ownerApi;
+        private OwnerContentApi ownerContentApi;
+        private OwnerProductApi ownerProductApi;
+        private CdnApi cdnApi;
+        private ConsumerClient consumerApi;
+        private ConsumerTypeApi consumerTypeApi;
+        private RulesApi rulesApi;
+        private RolesApi rolesApi;
+        private UsersApi usersApi;
+        private JobsClient jobsApi;
+
+        private OwnerDTO owner;
+        private ConsumerDTO consumer;
+        private CdnDTO cdn;
+        private ZipFile export;
+        private File manifest;
+        private Map<String, ProductDTO> productIdToProduct = new HashMap<>();
+
+        @BeforeAll
+        void beforeAll() throws Exception {
+            client = ApiClients.admin();
+            ownerApi = client.owners();
+            ownerContentApi = client.ownerContent();
+            ownerProductApi = client.ownerProducts();
+            cdnApi = client.cdns();
+            consumerApi = client.consumers();
+            consumerTypeApi = client.consumerTypes();
+            rulesApi = client.rules();
+            rolesApi = client.roles();
+            usersApi = client.users();
+            jobsApi = client.jobs();
+
+            initializeData();
+
+            manifest = createExport(client, consumer.getUuid(), cdn);
+            export = ExportUtil.getExportArchive(manifest);
+        }
+
+        @AfterAll
+        void afterAll() throws Exception {
+            export.close();
+        }
+
+        @Test
+        @DisplayName("should export consumer types")
+        void shouldExportConsumerTypes() throws Exception {
```

Comment:
This test is backed by the utility method which makes very strong assumptions about files never having conflicting names between directories. We should be pulling very specific files or file data on an individual basis to verify it exists where and how we expect it. For example, I'd expect this test to look something closer to the following:

```
List<ConsumerTypeDTO> types = this.consumerTypeApi.getConsumerTypes();
for (ConsumerTypeDTO ctype : types) {
  ZipEntry entry = this.export.getEntry("path/to/types/" + ctype.getName() + ".json");
  assertNotNull(entry);

  try (InputStream istream = this.export.getInputStream(entry)) {
    ConsumerTypeDTO parsed = ApiClient.MAPPER.readValue(istream, ConsumerTypeDTO.class);
    assertEquals(ctype, parsed);
  }
}
```

Bonus points for also verifying that there aren't additional files left in the directory that contains the JSON files for consumer types.

Reply:
Updated